libs/ui-watt/src/lib/components/table/watt-table-data-source.ts

```diff
@@ -14,9 +14,93 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { DataSource } from '@angular/cdk/collections';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
 import { MatTableDataSource } from '@angular/material/table';
+import { WattPaginatorComponent } from '../paginator';
 
 /**
  * @see https://material.angular.io/components/table/api#MatTableDataSource
  */
-export class WattTableDataSource<T> extends MatTableDataSource<T> {}
+export class WattTableDataSource<T> extends DataSource<T> {
+  private dataSource = new MatTableDataSource<T>();
+  private _paginator: MatPaginator | WattPaginatorComponent | null = null;
+
+  constructor(data?: T[]) {
+    super();
+    if (data) this.dataSource.data = data;
+  }
+
+  get data() {
+    return this.dataSource.data;
+  }
+  set data(data: T[]) {
+    this.dataSource.data = data;
+  }
+
+  get filter() {
+    return this.dataSource.filter;
+  }
+  set filter(filter: string) {
+    this.dataSource.filter = filter;
+  }
+
+  get filteredData() {
+    return this.dataSource.filteredData;
+  }
+  set filteredData(filteredData: T[]) {
+    this.dataSource.filteredData = filteredData;
+  }
+
+  get paginator() {
+    return this._paginator;
+  }
+  set paginator(paginator: MatPaginator | WattPaginatorComponent | null) {
+    this._paginator = paginator;
+    this.dataSource.paginator =
+      paginator instanceof WattPaginatorComponent
+        ? paginator.instance
+        : paginator;
+  }
+
+  get sort() {
+    return this.dataSource.sort;
+  }
+  set sort(sort: MatTableDataSource<T>['sort']) {
+    this.dataSource.sort = sort;
+  }
+
+  get filterPredicate() {
+    return this.dataSource.filterPredicate;
+  }
+  set filterPredicate(
+    filterPredicate: MatTableDataSource<T>['filterPredicate']
+  ) {
+    this.dataSource.filterPredicate;
+  }
+
+  get sortData() {
+    return this.dataSource.sortData;
+  }
+  set sortData(sortData: MatTableDataSource<T>['sortData']) {
+    this.dataSource.sortData;
```

Comment:
question: shouldn't `sortData` be asigned to `this.dataSource.sortData`?

Reply:
Whoops, good catch!